### ISSUE TITLE
Preload/Download files so they are available when offline

### ISSUE THREAD

**Issue by philippzentner** [Stremio/stremio-addon-sdk#132]
Hey,

it would be great if we could preload/download files so that they are available offline.
How would you approach that when writing a plugin?

Best

**Comment by jaruba** [comment]
You don't, downloading files is a client feature. Stremio Desktop has a button when you right click a video to allow downloading files. Otherwise, there's also a community project that allows downloading and managing files from Stremio: https://github.com/BurningSands70/stremio-downloader

So this has nothing to do with the SDK itself.

**Comment by philippzentner** [comment]
@jaruba This is really not what I mean. Instead of opening a browser it should:
* Be a simple **right-click->download** on the episode-name
* Have a status indicator within the episode-list that it is: downloading or downloaded
* Automatically saves the file at a specific place (can be adjusted in settings but can really just be in a temp/cache directory)
* After having watched the movie it should ask me if I want to delete the file to free up space

**Comment by jaruba** [comment]
@philippzentner 
What you describe sounds a lot like the Stremio Downloader community project: https://github.com/BurningSands70/stremio-downloader

An in-built feature like you are describing has been requested many times before, Stremio was built for streaming though, it's not a downloader. We have debated such a feature in the past though, but it's much more complicated then you would presume. We have various apps: Win, OSX, Linux, Android, Web. (and probably many more to come)

For the PC version this is easier then for others. But what about Android that has little free space or in Android TV cases very very little free space on the device. And what about Web? Which can't manage downloads at all in the way you are saying.

This doesn't mean we won't do it though, we've discussed this feature at length in the past, but keep in mind that our primary tasks are making Stremio available on more OSs, ensuring a great playback experience and making the apps more flexible for community developers (so more / better addons can be built). Everything else is secondary and we cannot give an ETA for them.
